### PR TITLE
Add junit report for racetest and fix test failures

### PR DESCRIPTION
**Pull request by howardjohn**
Please provide a description for what this PR is for.

This should allow prow to report which tests failed, currently it just says it failed and you have to find the failure amongst thousands of logs

[ ] Configuration Infrastructure
[ ] Docs
[ ] Installation
[ ] Networking
[ ] Performance and Scalability
[ ] Policies and Telemetry
[ ] Security
[x] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

### CODE REVIEW

**Comment by howardjohn** [comment]
Related to https://github.com/istio/istio/issues/15121

File: pkg/kube/secretcontroller/secretcontroller_test.go [hunk 1]
```diff
@@ -83,13 +84,13 @@ func mockCreateInterfaceFromClusterConfig(_ *clientcmdapi.Config) (kubernetes.In
 
 func verifyControllerDeleted(t *testing.T, timeoutName string) {
 	pkgtest.NewEventualOpts(10*time.Millisecond, 5*time.Second).Eventually(t, timeoutName, func() bool {
-		return testDeleteControllerCalled == true
+		return atomic.AddInt32(&testDeleteControllerCalled, 0) == 1
```
Comment (ozevren) [inline pkg/kube/secretcontroller/secretcontroller_test.go]: use atomic.LoadInt32?

Comment (howardjohn) [reply pkg/kube/secretcontroller/secretcontroller_test.go]: read the docs wrong and missed that, thanks

**Pull request merged**